### ISSUE TITLE
Typescript Typegen fail to generate component type if colon is present in field name

### ISSUE THREAD

**Issue by Killian-G** [strapi/strapi#22051]
### Required System information

- Node.js version: v20.17
- PNPM version: 9.10.0
- Strapi version: 5.2.0
- Database: sqlite
- Operating system: Linux Ubunutu 22.04
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When a component has a colon in one of its field names, if you generate type with the command `pnpm exec strapi ts:generate-types` you will get an error during type generation

### Steps to reproduce the behavior

1. Create a component
2. Add a field with a name like `og:description`
3. Save component
4. Generate type with `pnpm exec strapi ts:generate-types`

### Expected behavior

The type generation should generate the types without errors

### Screenshots

![image](https://github.com/user-attachments/assets/27ca0e72-252e-459f-9d66-dffe394b3499)

### Additional context

I think you should add quotes around attribute properties in the typescript generated file when a colon is present in the field name

**Comment by derrickmehaffy** [comment]
It's not just types that using `:` in a field name violates our field name limits. Did you set this in the content-type build or manually in the schema.json file?

**Comment by Killian-G** [comment]
I am using your SEO plugin, which has these field names with a colon in their components.
In my project, I was able to fix the issue, by patching your internal package @strapi/typescript-utils, which was responsible for Typescript codegen, by using string literal instead of identifier when a property name as some special characters that are not allowed without quotes, I can do a PR for this issue with my patch if you want

**Comment by nick-ma** [comment]
> I am using your SEO plugin, which has these field names with a colon in their components. In my project, I was able to fix the issue, by patching your internal package @strapi/typescript-utils, which was responsible for Typescript codegen, by using string literal instead of identifier when a property name as some special characters that are not allowed without quotes, I can do a PR for this issue with my patch if you want

Could you kindly share where and how to do the patch?

**Comment by draghia-andrei** [comment]
I managed to patch it in node_modules\@strapi\typescript-utils\lib\generators\common\models\attributes.js
Patched function: attributeToPropertySignature

Patched version:
`const attributeToPropertySignature = (schema, attributeName, attribute) => {
  const baseType = getAttributeType(attributeName, attribute, schema.uid);

  if (baseType === null) {
    return null;
  }

  const modifiers = getAttributeModifiers(attribute);

  const nodes = [baseType, ...modifiers];

  const isValidIdentifier = /^[a-zA-Z_$][a-zA-Z0-9_$]*$/.test(attributeName);

  const nameNode = isValidIdentifier
    ? factory.createIdentifier(attributeName)
    : factory.createStringLiteral(attributeName);

  return factory.createPropertySignature(
    undefined,
    nameNode,
    undefined,
    factory.createIntersectionTypeNode(nodes)
  );
};`

Basically if it's not a valid identifier, put it as a string literal, otherwise keep it as identifier.

**Comment by unrevised6419** [comment]
> I am using your SEO plugin, which has these field names with a colon in their components.

Hey, @Killian-G, can you please provide the link to what SEO plugin are you using?
The Strapi SEO plugin does not have any attribute name with a colon in it.

PS: I suppose this issue can be closed as it is not reproducible.
